wpt_common.py

```diff
@@ -141,3 +141,107 @@ def fetch_all_prs():
 
 def is_export_pr(pr):
     return bool(pr['chromium_commit'])
+
+
+def pr_number_from_tag(tagish):
+    """Extracts the PR number as integer from a string that begins with
+    merge_pr*, like merge_pr_6581 or merge_pr_6589-1-gc9db8d86f6."""
+    match = re.search(r'merge_pr_(\d+)', tagish)
+    if match is not None:
+        return int(match.group(1))
+    return None
+
+
+def get_tagged_prs():
+    """Gets the set of integers for which merge_pr_* exists."""
+    # --format="%(refname:strip=2) %(objectname)" would also include SHA-1
+    output = wpt_git(['tag', '--list', 'merge_pr_*'])
+    return set(pr_number_from_tag(tag) for tag in output.split() if tag)
+
+
+def git_contained_pr(commit):
+    """Returns the number of the latest contained PR of commit using git describe."""
+    try:
+        tag = wpt_git(['describe', '--tags', '--match', 'merge_pr_*', commit])
+        return pr_number_from_tag(tag)
+    except subprocess.CalledProcessError:
+        return None
+
+
+def pr_number(pr):
+    return int(pr['PR'])
+
+
+def pr_date(pr):
+    return dateutil.parser.parse(pr['merged_at'])
+
+
+def get_pr_latencies(prs, events=None, event_sha=None, event_date=None):
+    """For each PR, find the earliest event that included that PR,
+    and calucate the latencies between the PR and the event.
+
+    Args:
+        prs: list of PRs from fetch_all_prs()
+        events: list of events in any format
+        event_sha: function to get a sha from an event
+        event_date: function to get a datetime.datetime from an event
```

Comment:
Nit: Perhaps `event_{sha,date}_func` to make the functors more obvious.

Reply:
OK!